File: BubbleSort.py
# -*- coding: utf-8 -*-
"""
Created on Mon Feb 13 12:09:06 2023

@author: Administrator
"""

# Bubble Sort  


myList =  [0,5,10,45,15,19,8,11,13,2] # 
upperBound = 9
lowerBound = 0
top =  upperBound
swap = False

while swap or top > 0:
    swap = False
    for index in range(top):
        if myList[index] > myList[index + 1]:
            temp = myList[index]
            myList[index] = myList[index+1]
            myList[index + 1] = temp
            swap = True
            
    top = top - 1
print(myList)

File: BubbleSort.vb
'VB program for bubble sort
Module Module1
    Sub Main()
        Dim myList() As Integer = New Integer() {70, 46, 43, 27, 57, 41, 45, 21, 14}
        Dim index, top, temp As Integer
        Dim swap As Boolean
        top = myList.Length - 1
        Do
            swap = False
            For index = 0 To top - 1 Step 1
                If myList(index) > myList(index + 1) Then
                    temp = myList(index)
                    myList(index) = myList(index + 1)
                    myList(index + 1) = temp
                    swap = True
                End If
            Next
            top = top - 1
        Loop Until (Not swap) Or (top = 0)
        'output the sorted array
        For index = 0 To myList.Length - 1
            Console.Write(myList(index) & " ")
        Next
        Console.ReadKey() 'wait for keypress

    End Sub
End Module